submissions/andrewklmn/js-movie-seat-booking/modules/ticket-updater.js

```diff
@@ -11,6 +11,78 @@ import {
 
 import getSelectedSeats from './seat-selector';
 
+const MAX_DETAILED_SEATS_IN_TICKET = 6;
+const MAX_LENGTH_OF_SIMPLE_SEAT_LIST = 4;
+
+function getShortedSeatsInRowRecord(seats) {
+  if (seats.length < MAX_LENGTH_OF_SIMPLE_SEAT_LIST) {
+    return seats.join(',');
+  }
+
+  const sequences = [[seats[0]]];
+  let currentSequenceIndex = 0;
+
+  let i = 1;
+  while (i < seats.length) {
+    if (seats[i] - seats[i - 1] === 1) {
+      sequences[currentSequenceIndex][1] = seats[i];
+    } else {
+      currentSequenceIndex += 1;
+      sequences.push([seats[i]]);
+    }
+    i += 1;
+  }
+
+  return sequences.map((sequence) => {
+    if (sequence.length === 1) {
+      return sequence[0];
+    }
+    if (sequence[1] - sequence[0] === 1) {
+      return sequence.join(',');
+    }
+    return sequence.join('-');
+  }).join(',');
+}
+
+function getSeatPosition(seatElement) {
+  const seatPosition = seatElement.id.split('_');
+  return {
+    row: seatPosition[1],
+    seat: seatPosition[2],
+  };
+}
+
+function drawTicketPosition(selectedSeats) {
+  if (selectedSeats.length <= MAX_DETAILED_SEATS_IN_TICKET) {
+    return selectedSeats.map((nextSeat) => {
+      const { row, seat } = getSeatPosition(nextSeat);
+      return `<li>row#${row} seat:${seat}</li>`;
+    }).join('');
+  }
+
+  const selectedRows = selectedSeats.reduce((acc, nextSeat) => {
+    const { row } = getSeatPosition(nextSeat);
+    if (!acc.includes(row)) {
+      acc.push(row);
+    }
+    return acc;
+  }, []);
+
+  let answer = '';
+
+  selectedRows.forEach((nextRow) => {
+    const seatsInRow = [];
+    selectedSeats.forEach((x) => {
+      const { row, seat } = getSeatPosition(x);
+      if (nextRow === row) {
+        seatsInRow.push(seat);
+      }
```

Comment:
forEach with push() in callback function you can replace with [].map(), but it's just my opinion, rewrite or not it's up to you)

Reply:
I thought about [].map() here, but this callback pushes only a few seats from 'selectedSeats' (array with all rows and seats) when  IF condition is true...  
So the length of the result array usualy less than the length of selectedSeats array...
This is why I selected forEach() with [].push() in callback instead.

Reply:
My bad I mean [].filter() )